plugins/catalog-backend/src/service/DefaultRefreshService.ts

```diff
@@ -26,23 +26,24 @@ export class DefaultRefreshService implements RefreshService {
 
   async refresh(options: RefreshOptions) {
     await this.database.transaction(async tx => {
-      const { entityRefs } = await this.database.listAncestors(tx, {
-        entityRef: options.entityRef,
-      });
-      const locationAncestor = entityRefs.find(ref =>
-        ref.startsWith('location:'),
-      );
+      const refreshEntityRef = async (entityRef: string) => {
+        const { entityRefs } = await this.database.listAncestors(tx, {
+          entityRef,
+        });
+
+        const locationAncestor = entityRefs.find(ref =>
+          ref.startsWith('location:'),
+        );
```

Comment:
```suggestion
        const locationAncestor = entityRefs.filter(ref =>
          ref.startsWith('location:'),
        );
```
I think we can just do something like this since we're fetching the full ancestry list?